### DISCUSSION TITLE
"Tabular" cells?

### DISCUSSION THREAD

**Discussion by stillyslalom** [fonsp/Pluto.jl#866] [Pluto development]
I've found myself staring at walls of single-line cells, wondering where all my screen space went.
![image](https://user-images.githubusercontent.com/4324760/105392410-c886a000-5be0-11eb-928d-7d659d1249e5.png)
It'd be great to have a tabular cell style to make these sorts of one-liner parameter entries more compact and easy to read - something like an n-row, three-column table where each row is its own Pluto cell, and the columns are | input | output | comments |, like
![image](https://user-images.githubusercontent.com/4324760/105396946-14881380-5be6-11eb-8b91-bd1481f41a8f.png)

**Comment by BrandonConder** [comment]
You can run more than one line with a `begin; end` block, but it won't display the output from each line.

This seems like a good opportunity to create a macro that converts a block into a more useful format (like the `@show` macro I've come to know and love for Jupyter...). Perhaps something like

```jl
@tabular begin
	τ = 2π  # The real circle constant!
	ω = τ .* f  # Angular frequency
end
```

Where the macro assigns each statement and returns a formatted markdown table, for example:

```jl
md"""
|Input         | Output    | Comments                  |
|:-------------|:----------|:--------------------------|
|τ = 2π        | $(τ)      | The real circle constant! |
|ω = τ .* f    | $(ω)      | Angular frequency         |
"""
```

Such that the cell renders like:

Input | Output | Comments
:-- | :-- | :--
τ = 2π | 6.283185307179586 | The real circle constant!
ω = τ .* f | 6283.185307179586 | Angular frequency

Some more thought would need to go into number formatting, but this all seems very possible ...at least with my limited understanding of macros.

**Reply by stillyslalom** [reply]
Smart approach, and easy to imagine doing with minimal modifications to Pluto. The only downside is the cost of updating all the cells that depend on any combination of parameters in the table whenever you update a single parameter, but that's easy enough for a user to avoid by splitting the block into multiple blocks as needed.

**Reply by fonsp** [reply]
This can be done inside Pluto today! With _no_ modifications to Pluto. That's the cool thing about Julia macros!
